### PR TITLE
Feature mailmotor one page subscribe widget

### PR DESCRIPTION
**Pull request by cirykpopeye**
## Type

> remove the types that don't apply

- Critical bugfix
- Feature

## Resolves the following issues

- When a faulty list id, or no users in a list it was throwing a foreach error, solved by checking with is_array()

## Pull request description

- One page submission on MailChimp implemented in the widget, instead of linking to the action.

### CODE REVIEW

File: src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig [hunk 1]
```diff
@@ -4,16 +4,15 @@
       <h3 class="title">{{ 'lbl.SubscribeToNewsletter'|trans|capitalize }}</h3>
     </div>
     <div class="bd">
-      <form action="{{ geturlforblock('Mailmotor', 'Subscribe' ) }}" method="post">
-        <input type="hidden" name="form" value="mailMotorSubscribeForm" />
+      {{ form_start(form) }}
+        {#<input type="hidden" name="form" value="mailMotorSubscribeForm" />#}
```
Comment (carakas) [inline src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig]: just remove it, don't leave code in comments

Comment (cirykpopeye) [inline src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig]: Removed. Forgot it was there, thanks!

File: src/Frontend/Modules/Mailmotor/Form/SubscribeType.php [hunk 2]
```diff
@@ -102,12 +106,14 @@ public function getInterests()
             $mailMotorInterests = $this->subscriber->getInterests();
 
             // Has interests
-            if (!empty($mailMotorInterests)) {
+            if (!empty($mailMotorInterests) && is_array($mailMotorInterests)) {
```
Comment (jeroendesloovere) [inline src/Frontend/Modules/Mailmotor/Form/SubscribeType.php]: Normally, we don't encourage to do two bugfixes in one PR.

Comment (carakas) [inline src/Frontend/Modules/Mailmotor/Form/SubscribeType.php]: But if you get errors along the way of fixing your initial bug, it is normal to also fix them

**Comment by jeroendesloovere** [comment]
I'm going to test this now and see if this a good solution

**Comment by jeroendesloovere** [comment]
Just tested this and it's not a required solution.

**Comment by carakas** [comment]
@jeroendesloovere the downside of that is that people have to fill in things on 2 pages, and for something as simple as this that is overkill afaik

**Comment by cirykpopeye** [comment]
Maybe we should have two seperate widgets for that then, or check if there are any interests to select?

**Comment by jeroendesloovere** [comment]
We just added a new bug. Can you look at the screenshot I provided above.
Then "submit" button is above the "interests".
@cirykpopeye can you fix this?

**Comment by tijsverkoyen** [comment]
@jeroendesloovere I'm not a usability-expert. But if you have a widget that states "Subscribe to the newsletter", with only the emailaddress field, that implies you only have to enter you email address. So entering extra details in the "next" step will have less conversion as a result.

File: src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig [hunk 3]
```diff
@@ -8,6 +8,9 @@
         <p>
           {{ form_row(form.email) }}
         </p>
+
+        {{ form_rest(form) }}
```
Comment (carakas) [inline src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig]: that won't work since it will also take the button with it

Comment (cirykpopeye) [inline src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig]: Yes indeed. Fixed.

File: src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig [hunk 4]
```diff
@@ -9,7 +9,7 @@
           {{ form_row(form.email) }}
         </p>
 
-        {{ form_rest(form) }}
+        {{ form_row(form.interests) }}
```
Comment (carakas) [inline src/Frontend/Modules/Mailmotor/Layout/Widgets/Subscribe.html.twig]: There are cases when the interests aren't added, this will now give an error in those cases.
You need to check if the field exists

File: src/Frontend/Modules/Mailmotor/Form/SubscribeType.php (hunk 2 above)
Comment (carakas) [inline src/Frontend/Modules/Mailmotor/Form/SubscribeType.php]: but it would indeed have been better to do it in a separate mr

Comment (cirykpopeye) [inline src/Frontend/Modules/Mailmotor/Form/SubscribeType.php]: I fixed it as I couldn't continue testing the feature. But indeed.
